### DISCUSSION TITLE
User's mine permissions not working

### DISCUSSION THREAD

**Discussion by buckhornmktg** [directus/directus#2827] [Archive]
When I create a new role and set the directus_users permission to "mine" only the current user is displayed in the user directory. When I create a new user with that role it doesn't show up in the user directory (or in relationship drop downs).

I have tried creating an owner field in the directus_users table. The owner id is not being set when a new user is created. 

I also added a user field and set it to owner. That will set the ID with a drop down, but the user directory still only display the current user. 

Am I doing something wrong?

**Comment by gunners81** [comment]
mine = owner (who created the record)
role = group of user

if you set as "mine" than only owner of the record can view/update/delete it

**Reply by buckhornmktg** [reply]
Exactly. 

Here's the problem I'm having... I have a user with an ID of 6. User 6 creates a new user with an ID of 7. User 6 is the owner of User 7. Only user 6 will show up in the user directory and in the drop down relationship lists. I have tried with a clean install. I have tried adding an owner field to the user's table and setting the owner ID with a hook. Nothing works. Only the current user is visible, not any other records owned by User 6.

**Reply by buckhornmktg** [reply]
Thanks for your help by the way and for updating with your solution on one of your own questions (that was very helpful).

**Reply by gunners81** [reply]
> Exactly.
> 
> Here's the problem I'm having... I have a user with an ID of 6. User 6 creates a new user with an ID of 7. User 6 is the owner of User 7. Only user 6 will show up in the user directory and in the drop down relationship lists. I have tried with a clean install. I have tried adding an owner field to the user's table and setting the owner ID with a hook. Nothing works. Only the current user is visible, not any other records owned by User 6.

hmm.. your hook to set owner ID works ?

**Reply by buckhornmktg** [reply]
Yes, the correct owner id is getting saved to the database with the hook. I'm getting the id using your example on the other thread you posted... acl->getUserId(). Works fine in that regard, but the owner id seems to be totally ignored by the access control.

The owner field type didn't automatically save the owner like it does on other collections either, hence the hook.

I could have sworn I tested this functionality a month or two ago on a clean install by just setting the permissions to mine and it worked as expected.

This was odd, because on a clean install, there's no owner id being stored in the user table, so I'm not sure why the option is there in the role permissions. I don't see any other fields like 'user' on the table that might store an owner id either.

I tried checking out a couple older versions with git. The behavior was the same with the past three releases. 

This all may be a moot point. The client is terrible. If I don't fire them, are you available to do some freelance development and/or to teach me some advanced Directus stuff? Or maybe you know someone if you're not interested?
